src/oci/oci_runtime.rs

```diff
@@ -0,0 +1,734 @@
+            ),
+            Create((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Delete(container_id) => {
+                self.build_cmd_vec(Vec::new(), Some(String::from(container_id)))
+            }
+            Events((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Exec((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Kill((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Pause(container_id) => self.build_cmd_vec(Vec::new(), Some(String::from(container_id))),
+            Ps((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Restore((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Resume(container_id) => {
+                self.build_cmd_vec(Vec::new(), Some(String::from(container_id)))
+            }
+            Run((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Start(container_id) => self.build_cmd_vec(Vec::new(), Some(String::from(container_id))),
+            State(container_id) => self.build_cmd_vec(Vec::new(), Some(String::from(container_id))),
+            Update((container_id, args)) => self.build_cmd_vec(
+                args.iter().map(ToString::to_string).collect(),
+                Some(String::from(container_id)),
+            ),
+            Init() => self.build_cmd_vec(Vec::new(), None),
+            List(args) => self.build_cmd_vec(args.iter().map(ToString::to_string).collect(), None),
+            Spec(args) => self.build_cmd_vec(args.iter().map(ToString::to_string).collect(), None),
+        }
+    }
+
+    /// Build a vec of `[command][args][container_id]`
+    fn build_cmd_vec(&self, args: Vec<String>, container_id: Option<String>) -> Vec<String> {
+        let mut res = vec![self.to_string()]
+            .into_iter()
+            .chain(args.into_iter())
+            .collect::<Vec<_>>();
+        if let Some(id) = container_id {
+            res.push(id)
+        }
+        res
+    }
+}
+
+impl fmt::Display for Subcommand {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_ref())
+    }
+}
```

Comment:
Can we use the strum derivation instead: https://docs.rs/strum/0.20.0/strum/derive.Display.html?

Reply:
I can try, when I was initially trying the above it always capitalized it. but I might have missed a setting. otherwise I would have too put `#[strum(serialize = "lowercase")]` above each enum value. and that seemed worse to me.

Reply:
ok. I figured this out. going to push it. I had to update the `cargo` file to include `strum_macro` and that odd part here was that there was not the same version of `strum_macro` and we are using strum. did not know if I should get the latest version or the one closest to the version of strum we are using. to be honest it did not seem to mater from what I could read but I'll let you decide.